File: what does the fox say/whatdoesthefoxsay.py
for i in range(int(input())):
    sounds = [w for w in input().split(" ")]
    known_sounds = set()
    while True:
        try:
            known_sounds.add(input().split(" goes ")[1])
        except:
            print(" ".join([w for w in sounds if w not in known_sounds]))

File: what does the fox say/whatdoesthefoxsay.js
let n = readline();

for (let i=0; i<n; i++) {
    let sounds = readline().split(" ");
    let line = "";
    let knownSounds = new Set();
    while (line !== "what does the fox say?") {
        line = readline();
        knownSounds.add(line.split(" goes ")[1]);
    }
    console.log(sounds.filter(s => !knownSounds.has(s)).join(" "));
}